resources/user.go

```diff
@@ -0,0 +1,212 @@
+// Mgmt
+// Copyright (C) 2013-2017+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"fmt"
+	errwrap "github.com/pkg/errors"
+	"log"
+	"os/user"
+)
+
+func init() {
+	RegisterResource("user", func() Res { return &UserRes{} })
+}
+
+//TODO
+type UserRes struct {
+	BaseRes `yaml:",inline"`
+	Gid     string `yaml:"gid"`
+	HomeDir string `yaml:"homedir"`
+	State   string `yaml:"state"` // state: exists, absent
+	Uid     string `yaml:"uid"`
+	exec    *ExecRes
```

Comment:
Probably also worth adding a newline between the end of the public fields and the start of the private ones.
For longer structs you might want to add newlines between different related groups of fields for example.